src/evergreen.ts

```diff
@@ -1,235 +1,262 @@
+   * @param verbose - returns task details rather than aggregated stats
+   * @param lookbackMins - look for tasks that ended at most this many minutes before now
+   * @param status - task statuses (can be comma-separated list) to filter on
+   * @returns nothing
+   */
+  public getRecentTasks(callback: request.RequestCallback, verbose?: boolean, lookbackMins?: number, status?: string) {
+    const params = {
+      verbose: verbose,
+      minutes: lookbackMins,
+      status: status,
+    };
+    this.getAPIResource(callback, apiV2Resource("status/recent_tasks"), params);
+  }
 
-    /**
-     * Gets the Evergreen banner message
-     *
-     * @param callback - function to process the response
-     * @returns nothing
-     */
-    public getBanner(callback: request.RequestCallback) {
-        this.getAPIResource(callback, apiV2Resource("admin/banner"));
-    }
+  /**
+   * Updates header cookies when given login credentials
+   *
+   * @param callback - function to process the response
+   * @param username - Evergreen user's username
+   * @param password - Evergreen user's password
+   * @returns nothing
+   */
+  public getToken(callback: request.RequestCallback, username?: string, password?: string) {
+    const params = {
+      username: username,
+      password: password,
+    };
+    this.postUIResource(callback, "login", params);
+  }
 
-    /**
-     * Sets the Evergreen banner message
-     *
-     * @param callback - function to process the response
-     * @param message - text to set in the banner
-     * @param theme - color theme to use
-     * @returns nothing
-     */
-    public setBanner(callback: request.RequestCallback, message: string, theme: string) {
-        const body = {
-            banner: message,
-            theme: theme,
-        };
-        this.postAPIResource(callback, apiV2Resource("admin/settings"), body);
-    }
+  /**
+   * Gets patches for a particular user
+   * @param callback - function to process the response
+   * @param username - Evergreen user's username
+   * @returns nothing
+   */
+  public getPatches(callback: request.RequestCallback, username?: string, page?: number) {
+    const resource = "json/patches/user/" + username;
+    const params = {
+      page: page,
+    };
+    this.getUIResource(callback, resource, params);
+  }
 
-    private formRequest(url: string, requireHeaders: boolean, body?: any): requestOpts {
-      let headers = {};
-      if (requireHeaders) {
-        headers = {
-          "Api-User": this.username,
-          "Api-Key": this.key,
-        };
-      }
-      const opts: requestOpts = {
-        headers: headers,
-        url: url,
+  /**
+   * Gets logs of a particular type for a particular task/
+   * @param callback - function to process the response
+   * @param id - identifier for the task whose logs we want
+   * @param type - type of log to return
+   * @returns nothing
+   */
+  public getLogs(callback: request.RequestCallback, url: string) {
```

Comment:
We shouldn't require the caller to know the URL. The function signature before was fine - would it work to just append the text=true?

Reply:
Yes, that works. I was messing around to see if using the URLs passed in the task response would work in this method, which they do, but I agree it makes more sense to not pass the entire URL.